### PR TITLE
es6: arrow functions

### PR DESCRIPTION
**Pull request by ikokostya**
https://github.com/ymaps/codestyle/issues/33

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -958,6 +959,65 @@ util.inherits(Stream, EventEmitter);
...
+
+  **Bad:**
+
+  ```js
+  [1, 2, 3].map(function (x) {
+      // ...
+  });
+  ```
+
+* Parentheses around arrow function parameters can be ommited.
```
Comment (ikokostya) [inline README.md]: I don't know which style is better. 
* Style without parentheses looks more clean, especially when multiple functions are chained together.
* Style with parentheses is consistent with cases when function takes zero or more than one parameters. Also in this style you need less typing when change number of function parameters.

Comment (tarmolov) [inline README.md]: I used to add parentheses for one parameter because of consistency but now I prefer to omit them for cleaner and more readable code :)

**Comment by tarmolov** [comment]
:+1:

**Comment by dmikis** [comment]
:+1:

**Comment by GREENpoint** [comment]
TypeScript style-guides quotes:

```
1. Use arrow functions over anonymous function expressions.
2. Only surround arrow function parameters when necessary. 
```
https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines

It's sound reasonable.

File: README.md [hunk 2]
```diff
@@ -958,6 +959,65 @@ util.inherits(Stream, EventEmitter);
 
 [⬆ back to TOC](#table-of-contents)
 
+### Arrow functions
+
+* [Arrow functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions)
+  should be used in place where callback function expected:
```
Comment (dodev) [inline README.md]: ... should be used where callback...

without `in place`

File: README.md [hunk 3]
```diff
@@ -958,6 +959,65 @@ util.inherits(Stream, EventEmitter);
...
+
+  > Explanation:
+  > * `this` keyword in arrow function binds to the upper scope's. It prevents usage invalid `this` or usage less
```
Comment (dodev) [inline README.md]: to the upper scope's `this`.

File: README.md [hunk 4]
```diff
@@ -958,6 +959,81 @@ util.inherits(Stream, EventEmitter);
 
 [⬆ back to TOC](#table-of-contents)
 
+### Arrow functions
+
+* [Arrow functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions)
+  should be used where anonymous function expected:
```
Comment (dodev) [inline README.md]: .. an annonymous function is expected..

**Comment by dodev** [comment]
:+1:

**Pull request merged**